### PR TITLE
fix(validations): align FE schemas with backend constraints

### PR DESCRIPTION
**Pull request by joe-bor**
## Summary
- Reduce username max length from 30 → 20 to match backend `@Size(max = 20)` constraint
- Add `max(255)` validation to event location field to match backend `@Size(max = 255)` constraint
- Add tests for location max length boundary

## Test plan
- [x] All 426 existing tests pass
- [x] New location boundary tests (255 accepted, 256 rejected)
- [x] Lint clean
- [x] Build succeeds

### CODE REVIEW

**Comment by joe-bor** [comment]
## PR Review: fix(validations): align FE schemas with backend constraints

### Summary
Aligns two frontend validation schemas with backend `@Size` constraints: reduces username max length from 30 → 20 and adds a 255-char max to event location. Includes boundary tests for the location change.

### Verdict: Approve

### Strengths
- **Minimal, focused changes** — only touches what's needed, no scope creep
- **Good boundary testing** — tests both 255 (accepted) and 256 (rejected) for location
- **Atomic commits** — one commit per schema change, clean conventional format
- **Error messages are clear** — user-friendly validation messages that state the actual limit

### Issues (Must Fix)
None.

### Suggestions (Non-blocking)

1. **Missing tests for username max length change** (`src/lib/validations/auth.ts`): The location change got boundary tests, but the username 30 → 20 change doesn't have corresponding test updates. Consider adding boundary tests (20 accepted, 21 rejected) for consistency — especially since this is a *tightening* of an existing constraint that could break users with 21-30 char usernames.

2. **Consider a migration/UX note**: If any existing users have usernames between 21-30 characters, they'll fail validation on their next edit. This is probably fine for FamilyHub's current state, but worth noting.

### Technical Debt
No new debt introduced. This PR *reduces* alignment debt between FE and BE validation rules.

File: src/lib/validations/auth.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const usernameSchema = z
     z
       .string()
       .min(3, "Username must be at least 3 characters")
-      .max(30, "Username must be 30 characters or less")
+      .max(20, "Username must be 20 characters or less")
```
Comment (joe-bor) [inline src/lib/validations/auth.ts:16]: Non-blocking suggestion: The location field got boundary tests (255 accepted, 256 rejected), but this max length reduction from 30 → 20 doesn't have corresponding test updates. Consider adding similar boundary tests for consistency — especially since this tightens an existing constraint.
